tools/plugins/com.liferay.ide.project.ui/src/com/liferay/ide/project/ui/migration/MigrateProjectHandler.java

```diff
@@ -161,11 +171,35 @@ public void beginTask( String taskName, int totalWork )
                         {
                             final List<Problem> problems = m.findProblems( location.toFile(), override );
 
-                            problems.addAll( problems );
+                            allProblems.addAll( problems );
                         }
                     }
 
-                    m.reportProblems( allProblems, Migration.DETAIL_LONG, "ide" );
+                    if( allProblems.size() > 0 )
+                    {
+                        CodeMigrationProblems cp = new CodeMigrationProblems();
+
+                        FileProblems fileProblems = new FileProblems();
+
+                        fileProblems.setProblems( allProblems );
+
+                        cp.setProblems( FileProblemsUtil.getFileProblemsArray( allProblems.toArray( new Problem[0] ) ).toArray(
```

Comment:
This is confusing, I think I know what it is doing but it isn't obvious.  Can you rewrite this section to make it clearer?